3MI0800270/2023-02-28/zad1.cpp

```diff
@@ -0,0 +1,80 @@
+#include <iostream>
+using namespace std;
+
+struct transition {
+    size_t to;
+    char c;
+};
+
+struct state {
+    transition transitions[4];
+    size_t num;
+};
+
+struct stateInfo {
+    state data[50];
+};
+
+bool hasTransition(const stateInfo& arg, size_t from, char c) {
+    size_t number = arg.data[from].num;
+    for (int j = 0;j < number;j++) {
+        if (arg.data[from].transitions[j].c == c) {
+            return true;
+        }
+    }
+}
+bool addTransition(stateInfo& arg, size_t from, char c, size_t to) {
```

Comment:
This hasn't really been covered yet (exceptions & error handling) but having a boolean return value to signal the success / failure of an operation (in this case - the initialization of an object) is not a good idea. Boolean values don't really mean much (if false nobody will be able to tell the tale of what exactly went wrong) and can easily be overlooked (e.g. if I don't do an explicit check about the given return value - I'll never now that an error occurred). Keep in mind that [some other mechanisms](https://www.tutorialspoint.com/cplusplus/cpp_exceptions_handling.htm) exist for doing just that :))